### PR TITLE
chore: Remove kubelet configuration from the v1 APIs

### PR DESCRIPTION
**Pull request by engedaam**
Fixes #N/A 

**Description**
- Remove the kubelet configuration field and validation
- V1 RFC: https://github.com/kubernetes-sigs/karpenter/pull/1222

**How was this change tested?**
- `make presubmit`

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: hack/validation/kubelet.sh
Comment (njtran) [inline hack/validation/kubelet.sh:1]: Just checking. Are these copy pasted?

**Review by njtran — COMMENTED** [review]
/lgtm 
/approve
/hold
Holding for response on the comment. Want to make sure this yq is copy pasted.

File: pkg/apis/crds/karpenter.sh_nodeclaims.yaml
Comment (njtran) [inline pkg/apis/crds/karpenter.sh_nodeclaims.yaml:1]: Do we need to separate the CRDs out by API version?

Comment (engedaam) [reply pkg/apis/crds/karpenter.sh_nodeclaims.yaml:1]: The yaml file contains both v1 and v1beta1. This is only removing the v1 kubelet config https://github.com/kubernetes-sigs/karpenter/blob/38350b856a44134537405214aa0e8b13e2682289/pkg/apis/crds/karpenter.sh_nodepools.yaml#L27

File: hack/validation/kubelet.sh
Comment (engedaam) [reply hack/validation/kubelet.sh:1]: Yes. We now only need kubelet validation for v1beta1

**Review by jonathan-innis — APPROVED** [review]
/lgtm
/approve

**Pull request merged**